### DISCUSSION TITLE
APIError: {'code': '42883', 'details': None, 'hint': 'No function matches the given name and argument types. You might need to add explicit type casts.', 'message': 'function http(unknown, text, unknown) does not exist'}

### DISCUSSION THREAD

**Discussion by bosoleil** [supabase/supabase#26871] [Questions] [answered]
I keep getting this error and I also have everything activated. but I can't find the error;

```
line 78, in execute
    raise APIError(r.json())
postgrest.exceptions.APIError: {'code': '42883', 'details': None, 'hint': 'No function matches the given name and argument types. You might need to add explicit type casts.', 'message': 'function http(unknown, text, unknown) does not exist'}
```

With this script:

```
import asyncio
import aiohttp
from supabase import create_client, Client

# Supabase setup
supabase_url = 'https://test.supabase.co'
supabase_key = 'APIKEY'
supabase: Client = create_client(supabase_url, supabase_key)

# Emailable API setup
emailable_api_key = 'APIKEY'
emailable_api_url = 'https://api.emailable.com/v1/verify'
![CleanShot 2024-05-29 at 16 34 14@2x](https://github.com/supabase/supabase/assets/365292/8f2b1e36-4ed9-4c0c-9a23-717cd6490e18)

async def verify_email(email):
    """Verify an email and return the response."""
    async with aiohttp.ClientSession() as session:
        params = {'email': email, 'api_key': emailable_api_key}
        async with session.get(emailable_api_url, params=params) as response:
            result = await response.json()
            print(f"Verifying {email}: {result}")  # Print the result for debugging
            return result

async def update_email_verification(email, verification_result):
    """Update the verification result in Supabase."""
    response = supabase.table('merged_asana_leads').update({'calliss': verification_result}).eq('email', email).execute()
    return response

async def main():
    # Fetch emails from Supabase
    result = supabase.table('merged_asana_leads').select('email').execute()
    emails = [email['email'] for email in result.data if 'email' in email]

    # Verify emails and update the results
    for email in emails:
        result = await verify_email(email)
        if 'state' in result:
            await update_email_verification(email, result['state'])
        else:
            print(f"Failed to verify {email}: {result}")  # Log failure if no 'state' in result

# Run the main function
loop = asyncio.get_event_loop()
loop.run_until_complete(main())

```
![CleanShot 2024-05-29 at 16 34 41@2x](https://github.com/supabase/supabase/assets/365292/5ea5e4e0-8d22-43c3-b08c-d4eaaa840d41)
![CleanShot 2024-05-29 at 16 34 56@2x](https://github.com/supabase/supabase/assets/365292/0b2acb93-566a-401f-89ab-dbbd95e3ad5f)

**Comment by GaryAustin1** [comment]
I'm not sure why you started a new post on the same issue we discussed earlier.  

Here you are not showing the function making the http call.

In this one https://github.com/orgs/supabase/discussions/26860 you show more of the error and it says what Postgres function the error is in.  

You need to find that function **validate_email()**  and show the code, or fix the http call in that function.  In your other post you showed a different function.  It looks like validate_email might be a trigger function you have being called from a table operation.

**Reply by GaryAustin1** [reply]
You should be able to look here:
![image](https://github.com/supabase/supabase/assets/54564956/df3ff6dc-326d-4d4c-80b2-24d6b045664a)
And find the function validate_email() (assuming it is still the same full error postgres error (check in the Postgre logs) as before.

**Reply by bosoleil** [reply]
Hey Gary - Thank you so much. 

I've posted again because I've decided to instead run that function from Python in my computer for more safety purposes and error handling. 

i do have a validate email function (that i can't delete) but when I run the python scrypt

![CleanShot 2024-05-29 at 17 41 43@2x](https://github.com/supabase/supabase/assets/365292/0d05ddc5-fe6b-44f1-bb63-72764d25dfc0)

this is my function:

```
DECLARE
  email_response JSONB;
BEGIN
  SELECT * INTO email_response
  FROM http(
    'GET',
    'https://api.emailable.com/v1/verify?email=' || NEW.email || '&api_key=live_test',
    '{}'
  );

  NEW.validation_response := email_response;
  RETURN NEW;
END;
```

But this is not the same error althought pretty similar. I just can't run an HTTP call for some reason. It's making me go crazy.
APIError(r.json())
postgrest.exceptions.APIError: {'code': '42883', 'details': None, 'hint': 'No function matches the given name and argument types. You might need to add explicit type casts.', 'message': 'function http(unknown, text, unknown) does not exist'}

**Reply by GaryAustin1** [reply, marked as answer]
I'm pretty sure it is the same error.  You are doing an operation on a table that causes that function to run.  It is probably on a trigger.  Look in the triggers in the dashboard.  You either need to fix that function or delete it and the trigger calling it.

You http call is not correct there.  https://github.com/pramsey/pgsql-http?tab=readme-ov-file#functions has the function details.

The format is:
![image](https://github.com/supabase/supabase/assets/54564956/7471d185-0171-4243-b4ca-22e4bc09881e)
So the 3rd parameter is an ARRAY of http_header objects.
Then you put the content type as text, then the body as text.
There is also a get version of the call `http_get` shown in the docs that is simpler.

**Reply by bosoleil** [reply]
BOOOM! What a champ!!!!!!!!!!!!!!!!!!!!!!!
